pkg/service/service.go

```diff
@@ -734,7 +755,7 @@ func (s *Service) upsertServiceIntoLBMaps(svc *svcInfo, onlyLocalBackends bool,
 	)
 
 	// Update sessionAffinity
-	if option.Config.EnableSessionAffinity {
+	if option.Config.EnableSessionAffinity && svc.l7LBProxyPort == 0 {
```

Comment:
Need more context here - This code block processes session affinity related code. Why is this `svc.l7LBProxyPort == 0` check added here?

Reply:
L7 LB shares the same union in the service map value, but as these features are not needed at the same time, L7 LB takes the precedence over session affinity. proxy port is non-zero when L7 LB is registered for this service.

I added utility function `isL7LBService()` and used it here. Will add a comment to the above effect as well.